tests/Integration/Behaviour/Features/Scenario/Currency/currency_management.feature

```diff
@@ -0,0 +1,49 @@
+# ./vendor/bin/behat -c tests/Integration/Behaviour/behat.yml -s currency
+@reset-database-before-feature
+Feature: Currency Management
+  PrestaShop allows BO users to manage currencies
+  As a BO user
+  I must be able to create, edit and delete currencies in my shop
+
+  Background:
+    Given shop "shop1" with name "test_shop" exists
+
+  Scenario: Adding and editing currency
+    When I add new currency "currency1" with following properties:
+      | iso_code         | EUR   |
+      | exchange_rate    | 0.88  |
+      | is_enabled       | 1     |
+      | shop_association | shop1 |
+    Then currency "currency1" should be "EUR"
+    And currency "currency1" exchange rate should be 0.88
```

Comment:
This obliges you to write 1 step for each currency property.
Did you consider using a TableNode for properties assertion as well, as in [CustomerManagerFeatureContext](https://github.com/PrestaShop/PrestaShop/blob/develop/tests/Integration/Behaviour/Features/Context/CustomerManagerFeatureContext.php#L183) ? That would allow you to do:
```
Then currency "currency1" should have the following properties:
      | iso_code               |   EUR  |
      | exchange_rate      |  0.88  |
      | is_enabled            |    true |
      | shop_association | shop1 |
```

Reply:
Yeah, but for me adding as separate step sounds more natural and debugging is a bit easier, but I can switch to `TableNode` if that's considered a better way.

Reply:
> but for me adding as separate step
That's true indeed. The compromise has to be done between how many lines you write VS how easy it is to read

Let's keep it implemented this way then 😉 however do you think we can try to create a generic step to be used for all these usecases ?

Something like `@Then currency "(.*)" "(.*)" should be "(.*)"
Param 1 : currency reference
Param 2: property name (name, exchange rate, is_enabled ...)
Param 3: property value

It would save us a lot o copy-paste steps

Reply:
Yeah, it would be nice, but it introduces other issues.

For example, `@Then currency "currency1" "status" should be "enabled"`, and we need to convert `disabled/enabled` to `true` or we should use `true/false` in step, but that would make it less readable. 

There are also other issues with field types I think.

Reply:
Indeed we'll need to add some logic, but it already started in [PrimitiveUtils](https://github.com/PrestaShop/PrestaShop/blob/develop/tests/Integration/Behaviour/Features/Context/Util/PrimitiveUtils.php)
See [PrimitiveUtils:: castStringBooleanIntoBoolean](https://github.com/PrestaShop/PrestaShop/blob/develop/tests/Integration/Behaviour/Features/Context/Util/PrimitiveUtils.php#L185) which is in charge of converting string values into the right boolean.

It can be triggered because we can look for the property being checked and get its type. So we know what type the property value should be and can cast it accordingly

Reply:
I can see how it would reduce amount of methods we need, but then we would have all complexity inside single method, which would perform checking if field exists, converting it's values and who knows what else. 

Now we very small method, that are easy to changed if ever needed.

Reply:
All right let's see how it evolves 😉the cool thing about tests is that SemVer does not apply so we can break anything later